bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -935,11 +940,27 @@ public void run() {
             while (true) {
                 // new journal file to write
                 if (null == logFile) {
+
                     logId = logId + 1;
+                    final JournalChannel.BufferedChannelBuilder bcBuilder;
+                    if (getDelay > 0 || addDelay > 0 || flushDelay > 0) {
```

Comment:
it would be better to use PowerMock and to not use this kind of "explict" instrumentation.
Recently we added a lot of test cases which tweak the journal

Reply:
will do.
PowerMock in tests that extends BookKeeperClusterTestCase a bit problematic, likely due to  
https://github.com/powermock/powermock/issues/822

Figured a way how to use just mockito there, will update PR soon.